src/MSALWrapper.Test/AuthFlow/AuthFlowsTest.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.MSALWrapper.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using FluentAssertions;
+
+    using Microsoft.Authentication.MSALWrapper.AuthFlow;
+
+    using NUnit.Framework;
+
+    internal class AuthFlowsTest
+    {
+        [Test]
+        public void Requires_Flows()
+        {
+            Action subject = () => new AuthFlows(Guid.NewGuid(), Guid.NewGuid(), null);
+            subject.Should().Throw<ArgumentNullException>().WithMessage("Value cannot be null. (Parameter 'authFlows')");
+        }
+
+        [Test]
+        public void AuthFlows_Has_Id()
+        {
+            var client = Guid.NewGuid();
+            var tenant = Guid.NewGuid();
+            var subject = new AuthFlows(client, tenant, new List<IAuthFlow>());
+            subject.Id.Should().Be($"{client}_{tenant}");
+        }
+
+        [Test]
+        public void Is_Iterable()

```

Comment:
We can probably write a test on `Is_Iterable` when `IEnumerable<IAuthFlow> flows` is not an empty list.

Reply:
The only reason I wrote this test was to really TDD it into existence. The fact that a LINQ query on an IEnumerable<T> works, tells us it's interable. This is a case of where adding more tests on it being iterable are testing DotNet more than our own code, and we don't need to test DotNet here :).